### PR TITLE
Fixes for systemv

### PR DESCRIPTION
**Pull request by sellis**
All of these changes were necessary for me to get the systemv redhat functionality to work properly.

###### lib/daemon.js
Escape the quotes around the systemv script args because they will later be put inside of another quoted string (su -c "command") in the init.d script, to support running the service as a specific user.

###### lib/systemd.js
Added stub/no-op functions to the systemd Service object so client code can call enable() and disable() without having to worry if is a systemd or systemv instance (avoids Typerror not a function in client code)

###### lib/systemv.js
Moved code that determines OS type to utility function since it now needs to be called from 3 places, generate(), enable(), and disable()

Made enable() and disable() work for redhat which needs to use chkconfig not update-rc.d

Filling in the opt object from the config properties had two incorrect property names.

###### lib/templates/redhat
The script now changes to the specific user account using su -c "command" and pipes the PID to the pidfile.

Now we just kill the daemon (no -9) to send SIGTERM instead of SIGKILL.

Now we read the PID to kill from the pidfile in the do_stop() instead of using pgrep. I found pgrep was returning multiple pids for our clustered system and not working properly.

Implemented the status() method.

###### lib/wrapper.js
In our clustered system, needed to handle SIGTERM in order for all the children of the wrapper process to die properly.

### CODE REVIEW

**Comment by sellis** [comment]
@coreybutler  Hi Corey, just checking to see if you saw this pull request. Thanks.

File: lib/systemv.js [hunk 1]
```diff
@@ -152,8 +161,8 @@ var init = function(config){
             description: config.description,
             user: config.user || 'root',
             group: config.group || 'root',
-            pidroot: config.pidroot || '/var/run',
-            logroot: config.logroot || '/var/log',
+            pidroot: config.piddir || '/var/run',
```
Comment (coreybutler) [inline lib/systemv.js]: Why do these need to be renamed? I worry it may break some existing code. Additionally, the configuration options are supposed to be consistent across the windows and mac version (not so relevant on windows).

File: lib/systemv.js [hunk 2]
```diff
@@ -152,8 +161,8 @@ var init = function(config){
...
+            logroot: config.logpath || '/var/log',
```
Comment (coreybutler) [inline lib/systemv.js]: Same question as the line above.

**Comment by coreybutler** [comment]
Thanks for the ping on this - I must have missed the notice. 

Most of these changes look fine. I did notice some renaming of configuration attributes that I'm a little concerned may break some implementations. Additionally, have you tested this in a non-redhat environment? Sadly, I never got unit tests running, but I'd still like to make sure ubuntu/debian users don't come across any weird edge cases.

**Comment by sellis** [comment]
Thanks for looking at the changes. I'm not renaming any properties, I'm just making the property names match what is documented on the incoming config object. Without my changes, client code setting those properties has no effect.

**Comment by sellis** [comment]
According to the docs: http://coreybutler.github.io/node-linux/manual/#!/api/nodelinux.Service

The properties coming in on the config object are piddir and logpath, hence the change is needed for the property values to be set properly on the opt object.

**Comment by coreybutler** [comment]
I'd prefer there be regression testing on non-RedHat systems, but since I don't have time to do that right now, I'll merge this. Thanks for the contribution!

**Comment by sellis** [comment]
Yeah sorry, I'd prefer to test on debian too, but don't currently have access to a machine. I will see about doing that next week. Thanks for accepting the pull request and thanks for a great node module!

**Comment by coreybutler** [comment]
BTW - this has been pushed to npm as v1.0.4. Thanks again!
